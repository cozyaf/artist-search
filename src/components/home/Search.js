import React from 'react';
import PropTypes from 'prop-types';

function Search({ text, handleChange, handleClick }) {
  return (
    <>
      <label>Search:
        <input name="text" type="text" placeholder="Search for artist..." value={text} onChange={handleChange}/>
      </label>
      <button onClick={handleClick}>Magic Time</button>
    </>
  );
}

Search.propTypes = {
  text: PropTypes.string.isRequired,
  handleChange: PropTypes.func.isRequired,
  handleClick: PropTypes.func.isRequired
};

export default Search;
