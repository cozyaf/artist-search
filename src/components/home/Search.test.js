import React from 'react';
import { shallow } from 'enzyme';
import Search from './Search';

describe('Search component', () => {
  it('renders Search', () => {
    const wrapper = shallow(<Search text="sup" handleChange={() => {}}  handleClick={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });
});
