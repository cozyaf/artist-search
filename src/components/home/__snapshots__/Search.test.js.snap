// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component renders Search 1`] = `
<Fragment>
  <label>
    Search:
    <input
      name="text"
      onChange={[Function]}
      placeholder="Search for artist..."
      type="text"
      value="sup"
    />
  </label>
  <button
    onClick={[Function]}
  >
    Magic Time
  </button>
</Fragment>
`;
